chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -251,33 +258,114 @@ impl NetworkState {
         false
     }
 
+    /// Register a direct connection to a new peer. This will be called after successfully
+    /// establishing a connection with another peer. It become part of the connected peers.
+    ///
+    /// To build new edge between this pair of nodes both signatures are required.
+    /// Signature from this node is passed in `edge_info`
+    /// Signature from the other node is passed in `full_peer_info.edge_info`.
+    pub async fn register(
+        self: &Arc<Self>,
+        clock: &time::Clock,
+        conn: Arc<connection::Connection>,
+    ) -> Result<(), RegisterPeerError> {
+        let this = self.clone();
+        let clock = clock.clone();
+        self.spawn(async move {
+            let peer_info = &conn.peer_info;
+            // Check if this is a blacklisted peer.
+            if peer_info.addr.as_ref().map_or(true, |addr| this.peer_store.is_blacklisted(addr)) {
+                tracing::debug!(target: "network", peer_info = ?peer_info, "Dropping connection from blacklisted peer or unknown address");
+                return Err(RegisterPeerError::Blacklisted);
+            }
+
+            if this.peer_store.is_banned(&peer_info.id) {
+                tracing::debug!(target: "network", id = ?peer_info.id, "Dropping connection from banned peer");
+                return Err(RegisterPeerError::Banned);
+            }
+
+            if conn.peer_type == PeerType::Inbound {
+                if !this.is_inbound_allowed(&peer_info) {
+                    // TODO(1896): Gracefully drop inbound connection for other peer.
+                    let tier2 = this.tier2.load();
+                    tracing::debug!(target: "network",
+                        tier2 = tier2.ready.len(), outgoing_peers = tier2.outbound_handshakes.len(),
+                        max_num_peers = this.max_num_peers.load(Ordering::Relaxed),
+                        "Dropping handshake (network at max capacity)."
+                    );
+                    return Err(RegisterPeerError::ConnectionLimitExceeded);
+                }
+            }
+            // Verify and broadcast the edge of the connection. Only then insert the new
+            // connection to TIER2 pool, so that nothing is broadcasted to conn.
+            this.add_edges(&clock, vec![conn.edge.clone()])
+                .await
+                .map_err(|_| RegisterPeerError::InvalidEdge)?;
```

Comment:
`.map_err(|_: ReasonForBan| RegisterPeerError::InvalidEdge)?;`

would be a bit more readable and future-proof

Reply:
done